File: 00. Estructura/01. Lenguajes de proposito general/03. Persistencia/01. Directorios/Python/Directorios.py
# Libreria para el manejo de archivos
import os

# Direccion absoluta (Desde el la unidad de disco duro)
direccion = r'C:\carpeta1\carpeta2\carpetaN\archivo.extensión'

# Direccion relativa (Desde la raiz del proyecto)
direccion = 'carpeta1/carpeta1/carpetaN/archivo.extensión'

# Obtener directorio actual
os.getcwd()     # Como String
os.getcwdb()    # Como objeto directorio

# Cambiar directorio actual
os.chdir(direccion)

# Verificar si una dirección es directorio
os.path.isdir(direccion)

# Verificar si una dirección es archivo
os.path.isfile(direccion)

# Verificar si existe un directorio o archivo
os.path.exists(direccion)

# Obtener nombre del directorio actual
os.path.basename(direccion)

# Listar nombres de directorios y archivos contenidos en el directorio actual
os.listdir()

# Listar nombres de directorios y archivos contenidos en el directorio especificado
os.listdir(direccion)

# Aceder a un directorio especifico
os.walk("nombre")

# Listar nombres de directorios contenidos en el directorio especificado
[filename for filename in os.listdir(direccion) if os.path.isdir(os.path.join(direccion,filename))]

# Listar nombres de archivos contenidos en el directorio especificado
[filename for filename in os.listdir(direccion) if os.path.isfile(os.path.join(direccion,filename))]

# Crear un nuevo directorio
os.mkdir('nombre')

# Renombrear direcctorio
os.rename('test','new_one')

# Remover archivo
os.remove('nombre.extensión')

# Remover directorio vacio
os.rmdir('nombre')

# Remover directorio con el contenido
import shutil
shutil.rmtree('nombre')

File: 00. Estructura/01. Lenguajes de proposito general/03. Persistencia/01. Directorios/JavaScript/Directorios.js

// Modulos
fs = require('fs');			// Modulo para el manejo de archivos
path = require('path');		// Modulo para el manejo de directorios

// Direcciones
direccion = 'C:\\carpeta\\carpeta2\\carpetaN\\archivo.extensión';	// Direccion absoluta (Desde el la unidad de disco duro)
direccion = './carpeta1/carpeta2/carpetaN/archivo.extensión';		// Direccion relativa (Desde el directorio actual) (Los dos puntos (..) significa obtener carpeta padre a la actual)

// Acceso
direccion 						// Obtener dirección especficada
path.join(direccion, '..');		// Obtener padre
path.join(direccion, 'nombre');	// Obtener hijo

// Creación
fs.mkdir(direccion, (error) =>{});					// Crear direcctorio		
fs.writeFile(direccion, contendio, (error) =>{});	// Crear archivo

// Modificación
fs.rename(direccionActual, direccionNuena, (error) =>{});	// Renombrar direcctorio		

// Eliminación
fs.unlink(direccion, (error) =>{});						// Remover archivo
fs.rmdir(direccion, (error) =>{})						// Remover directorio vacio
fs.rmdir(direccion, {recursive: true}, (error) =>{});	// Remover directorio con el contenido

// Infromación
fs.lstatSync(direccion).isDirectory();	// Verificar si es la ruta es un direcctorio
fs.lstatSync(direccion).isFile();		// Verificar si es la ruta es un archivo
fs.access(direccion, (error) =>{});		// Verificar si la ruta existe
path.basename(direccion); 				// Obtener nombre del directorio o archivo
fs.readdirSync(direccion);				// Listar nombres de directorios y archivos contenidos en el directorio especificado


